db/metamapper.php

```diff
@@ -0,0 +1,29 @@
+<?php
+/**
+ * Nextcloud - Notes
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ */
+
+namespace OCA\Notes\Db;
+
+use OCP\IDBConnection;
+use OCP\AppFramework\Db\Mapper;
+
+class MetaMapper extends Mapper {
+
+	public function __construct(IDBConnection $db) {
+		parent::__construct($db, 'notes_meta');
+	}
+
+	public function getAll($userId) {
+		$sql = 'SELECT * FROM `*PREFIX*notes_meta` WHERE user_id=?';
+		return $this->findEntities($sql, [$userId]);
+	}
+
+	public function get($userId, $fileId) {
+		$sql = 'SELECT * FROM `*PREFIX*notes_meta` WHERE user_id=? AND file_id=?';
+		return $this->findEntity($sql, [$userId, $fileId]);
+	}
```

Comment:
So `last-modified` inside the response header and `mtime` of the actual note (file) are independent values, right? The header value will be stored in the database and the mtime is directly set on the note.

But would it hurt to generate the ETag for every response instead of caching it in the database?